scipy/optimize/_lbfgsb_py.py

```diff
@@ -279,24 +279,25 @@ def _minimize_lbfgsb(fun, x0, args=(), jac=None, bounds=None,
     x0 = asarray(x0).ravel()
     n, = x0.shape
 
+    # historically old-style bounds were/are expected by lbfgsb.
+    # That's still the case but we'll deal with new-style from here on,
+    # it's easier
     if bounds is None:
-        bounds = [(None, None)] * n
-    if len(bounds) != n:
-        raise ValueError('length of x0 != length of bounds')
-
-    # unbounded variables must use None, not +-inf, for optimizer to work properly
-    bounds = [(None if l == -np.inf else l, None if u == np.inf else u) for l, u in bounds]
-    # LBFGSB is sent 'old-style' bounds, 'new-style' bounds are required by
-    # approx_derivative and ScalarFunction
-    new_bounds = old_bound_to_new(bounds)
+        bounds = np.empty((2, n), dtype=float)
+        bounds[0] = -np.inf
+        bounds[1] = np.inf
+    else:
+        if len(bounds) != n:
+            raise ValueError('length of x0 != length of bounds')
+        bounds = np.array(old_bound_to_new(bounds))
```

Comment:
Would it be possible to rename `old_bound_to_new` to specify the adaptation that is made?

Reply:
going to leave as-is, this would be a much wider refactor.